frontend/src/metabase/visualizations/echarts/cartesian/chart-measurements/index.ts

```diff
@@ -186,29 +185,47 @@ export const getTicksDimensions = (
       ) + CHART_STYLE.axisTicksMarginY;
   }
 
+  const currentBoundaryWidth =
+    chartWidth -
+    CHART_STYLE.padding.x * 2 -
+    ticksDimensions.yTicksWidthLeft -
+    ticksDimensions.yTicksWidthRight;
+
+  ticksDimensions.minXTickSpacing = getMinXTickSpacing(
+    chartModel.dataset,
+    settings,
+    currentBoundaryWidth,
+    chartModel.xAxisModel.formatter,
```

Comment:
There is a TS error here

Reply:
It's on the `echarts` branch, doesn't come from this PR.